h/static/scripts/vendor/jquery.scrollintoview.js

```diff
@@ -109,29 +109,37 @@
 
 				var animOptions = {};
 
+				var yDir = "none";
```

Comment:
The fact that my assumption would be false is exactly why this is a bad idea. It's a reasonable assumption, and one that I think most developers would make. The fact we're violating that assumption is precisely what I'm objecting to.

If we've done it with vendored jsChannel, then it was a bad idea then too, in my opinion.

The issue I have is not with modifying library code. The issue is the manner in which it's done. Any of the following would be better alternatives:

- an explicit patch step when we build, with the differences maintained as a patch file
- we monkeypatch the library at runtime
- we move the library out of `vendor/` so that it's clear that we've forked it and are responsible for maintaining it
- we fork the library ourselves and maintain our patches in the fork